hail/python/dev/constraints.txt

```diff
@@ -0,0 +1 @@
+notebook>6,<7
```

Comment:
Another option here would be to just put a requirement in `hail/dev/requirements.txt`.